src/Compilers/Core/AnalyzerDriver/AnalyzerManager.cs

```diff
@@ -142,6 +142,30 @@ internal class AnalyzerManager
         }
 
         /// <summary>
+        /// Returns true if analyzer registered a compilation start action during <see cref="DiagnosticAnalyzer.Initialize(AnalysisContext)"/>
+        /// which registered a compilation end action and at least one other analyzer action, that the end action depends upon.
+        /// </summary>
+        public async Task<bool> GetAnalyzerHasDependentCompilationEndAsync(DiagnosticAnalyzer analyzer, AnalyzerExecutor analyzerExecutor)
```

Comment:
Well, yes we can by cloning the compilation start action execution in both the drivers. I can the move the per-compilation actions map out of AnalyzerManager into the IDE driver and compiler driver can execute them for compilation start action events.
Just takes the *goodness* of one single component/API handling the "analyzer -> all analyzer actions" mapping. I can go that route if that seems preferable, but eventually this API and IDE driver have to go away, so I don't see an advantage of one approach over another either way.